File: src/emu/cpu_opcodes.js
export const BRK     = 0x00

export const PLA     = 0x68
export const PLP     = 0x28
export const PHA     = 0x48
export const PHP     = 0x08

export const JMP_ABS = 0x4c
export const JMP_IND = 0x6c
export const BPL     = 0x10
export const BMI     = 0x30
export const BVC     = 0x50
export const BVS     = 0x70
export const BCC     = 0x90
export const BCS     = 0xb0
export const BNE     = 0xd0
export const BEQ     = 0xf0

export const JSR     = 0x20
export const RTI     = 0x40
export const RTS     = 0x60

export const LDA_IMM = 0xa9
export const LDA_ZER = 0xa5
export const LDA_ZRX = 0xb5
export const LDA_ABS = 0xad
export const LDA_ABX = 0xbd
export const LDA_ABY = 0xb9
export const LDA_PTX = 0xa1
export const LDA_PTY = 0xb1

export const LDX_IMM = 0xa2
export const LDX_ZER = 0xa6
export const LDX_ZRY = 0xb6
export const LDX_ABS = 0xae
export const LDX_ABY = 0xbe

export const LDY_IMM = 0xa0
export const LDY_ZER = 0xa4
export const LDY_ZRX = 0xb4
export const LDY_ABS = 0xac
export const LDY_ABX = 0xbc

export const STA_ZER = 0x85
export const STA_ZRX = 0x95
export const STA_ABS = 0x8d
export const STA_ABX = 0x9d
export const STA_ABY = 0x99
export const STA_PTX = 0x81
export const STA_PTY = 0x91

export const STX_ZER = 0x86
export const STX_ZRY = 0x96
export const STX_ABS = 0x8e

export const STY_ZER = 0x84
export const STY_ZRX = 0x94
export const STY_ABS = 0x8c

export const AND_IMM = 0x29
export const AND_ZER = 0x25
export const AND_ZRX = 0x35
export const AND_ABS = 0x2d
export const AND_ABX = 0x3d
export const AND_ABY = 0x39
export const AND_PTX = 0x21
export const AND_PTY = 0x31

export const ORA_IMM = 0x09
export const ORA_ZER = 0x05
export const ORA_ZRX = 0x15
export const ORA_ABS = 0x0d
export const ORA_ABX = 0x1d
export const ORA_ABY = 0x19
export const ORA_PTX = 0x01
export const ORA_PTY = 0x11

export const EOR_IMM = 0x49
export const EOR_ZER = 0x45
export const EOR_ZRX = 0x55
export const EOR_ABS = 0x4d
export const EOR_ABX = 0x5d
export const EOR_ABY = 0x59
export const EOR_PTX = 0x41
export const EOR_PTY = 0x51

export const ADC_IMM = 0x69
export const ADC_ZER = 0x65
export const ADC_ZRX = 0x75
export const ADC_ABS = 0x6d
export const ADC_ABX = 0x7d
export const ADC_ABY = 0x79
export const ADC_PTX = 0x61
export const ADC_PTY = 0x71

export const SBC_IMM = 0xe9
export const SBC_ZER = 0xe5
export const SBC_ZRX = 0xf5
export const SBC_ABS = 0xed
export const SBC_ABX = 0xfd
export const SBC_ABY = 0xf9
export const SBC_PTX = 0xe1
export const SBC_PTY = 0xf1

export const CMP_IMM = 0xc9
export const CMP_ZER = 0xc5
export const CMP_ZRX = 0xd5
export const CMP_ABS = 0xcd
export const CMP_ABX = 0xdd
export const CMP_ABY = 0xd9
export const CMP_PTX = 0xc1
export const CMP_PTY = 0xd1

export const CPX_IMM = 0xe0
export const CPX_ZER = 0xe4
export const CPX_ABS = 0xec

export const CPY_IMM = 0xc0
export const CPY_ZER = 0xc4
export const CPY_ABS = 0xcc

export const INX     = 0xe8
export const DEX     = 0xca
export const INY     = 0xc8
export const DEY     = 0x88

export const INC_ZER = 0xe6
export const INC_ZRX = 0xf6
export const INC_ABS = 0xee
export const INC_ABX = 0xfe

export const DEC_ZER = 0xc6
export const DEC_ZRX = 0xd6
export const DEC_ABS = 0xce
export const DEC_ABX = 0xde

export const ASL_IMP = 0x0a
export const ASL_ZER = 0x06
export const ASL_ZRX = 0x16
export const ASL_ABS = 0x0e
export const ASL_ABX = 0x1e

export const LSR_IMP = 0x4a
export const LSR_ZER = 0x46
export const LSR_ZRX = 0x56
export const LSR_ABS = 0x4e
export const LSR_ABX = 0x5e

export const ROL_IMP = 0x2a
export const ROL_ZER = 0x26
export const ROL_ZRX = 0x36
export const ROL_ABS = 0x2e
export const ROL_ABX = 0x3e

export const ROR_IMP = 0x6a
export const ROR_ZER = 0x66
export const ROR_ZRX = 0x76
export const ROR_ABS = 0x6e
export const ROR_ABX = 0x7e

export const BIT_ZER = 0x24
export const BIT_ABS = 0x2c

export const TXA     = 0x8a
export const TAX     = 0xaa
export const TYA     = 0x98
export const TAY     = 0xa8
export const TXS     = 0x9a
export const TSX     = 0xba

export const CLC     = 0x18
export const SEC     = 0x38
export const CLI     = 0x58
export const SEI     = 0x78
export const CLD     = 0xd8
export const SED     = 0xf8
export const CLV     = 0xb8

export const NOP     = 0xea
export const NOP_2   = 0x1a
export const NOP_3   = 0x3a
export const NOP_4   = 0x5a
export const NOP_5   = 0x7a
export const NOP_6   = 0xda
export const NOP_7   = 0xfa
export const NOP_8   = 0x80

File: rust/src/cpu_opcodes.rs
pub const BRK     : u8 = 0x00;

pub const PLA     : u8 = 0x68;
pub const PLP     : u8 = 0x28;
pub const PHA     : u8 = 0x48;
pub const PHP     : u8 = 0x08;

pub const JMP_ABS : u8 = 0x4c;
pub const JMP_IND : u8 = 0x6c;
pub const BPL     : u8 = 0x10;
pub const BMI     : u8 = 0x30;
pub const BVC     : u8 = 0x50;
pub const BVS     : u8 = 0x70;
pub const BCC     : u8 = 0x90;
pub const BCS     : u8 = 0xb0;
pub const BNE     : u8 = 0xd0;
pub const BEQ     : u8 = 0xf0;

pub const JSR     : u8 = 0x20;
pub const RTI     : u8 = 0x40;
pub const RTS     : u8 = 0x60;

pub const LDA_IMM : u8 = 0xa9;
pub const LDA_ZER : u8 = 0xa5;
pub const LDA_ZRX : u8 = 0xb5;
pub const LDA_ABS : u8 = 0xad;
pub const LDA_ABX : u8 = 0xbd;
pub const LDA_ABY : u8 = 0xb9;
pub const LDA_PTX : u8 = 0xa1;
pub const LDA_PTY : u8 = 0xb1;

pub const LDX_IMM : u8 = 0xa2;
pub const LDX_ZER : u8 = 0xa6;
pub const LDX_ZRY : u8 = 0xb6;
pub const LDX_ABS : u8 = 0xae;
pub const LDX_ABY : u8 = 0xbe;

pub const LDY_IMM : u8 = 0xa0;
pub const LDY_ZER : u8 = 0xa4;
pub const LDY_ZRX : u8 = 0xb4;
pub const LDY_ABS : u8 = 0xac;
pub const LDY_ABX : u8 = 0xbc;

pub const STA_ZER : u8 = 0x85;
pub const STA_ZRX : u8 = 0x95;
pub const STA_ABS : u8 = 0x8d;
pub const STA_ABX : u8 = 0x9d;
pub const STA_ABY : u8 = 0x99;
pub const STA_PTX : u8 = 0x81;
pub const STA_PTY : u8 = 0x91;

pub const STX_ZER : u8 = 0x86;
pub const STX_ZRY : u8 = 0x96;
pub const STX_ABS : u8 = 0x8e;

pub const STY_ZER : u8 = 0x84;
pub const STY_ZRX : u8 = 0x94;
pub const STY_ABS : u8 = 0x8c;

pub const AND_IMM : u8 = 0x29;
pub const AND_ZER : u8 = 0x25;
pub const AND_ZRX : u8 = 0x35;
pub const AND_ABS : u8 = 0x2d;
pub const AND_ABX : u8 = 0x3d;
pub const AND_ABY : u8 = 0x39;
pub const AND_PTX : u8 = 0x21;
pub const AND_PTY : u8 = 0x31;

pub const ORA_IMM : u8 = 0x09;
pub const ORA_ZER : u8 = 0x05;
pub const ORA_ZRX : u8 = 0x15;
pub const ORA_ABS : u8 = 0x0d;
pub const ORA_ABX : u8 = 0x1d;
pub const ORA_ABY : u8 = 0x19;
pub const ORA_PTX : u8 = 0x01;
pub const ORA_PTY : u8 = 0x11;

pub const EOR_IMM : u8 = 0x49;
pub const EOR_ZER : u8 = 0x45;
pub const EOR_ZRX : u8 = 0x55;
pub const EOR_ABS : u8 = 0x4d;
pub const EOR_ABX : u8 = 0x5d;
pub const EOR_ABY : u8 = 0x59;
pub const EOR_PTX : u8 = 0x41;
pub const EOR_PTY : u8 = 0x51;

pub const ADC_IMM : u8 = 0x69;
pub const ADC_ZER : u8 = 0x65;
pub const ADC_ZRX : u8 = 0x75;
pub const ADC_ABS : u8 = 0x6d;
pub const ADC_ABX : u8 = 0x7d;
pub const ADC_ABY : u8 = 0x79;
pub const ADC_PTX : u8 = 0x61;
pub const ADC_PTY : u8 = 0x71;

pub const SBC_IMM : u8 = 0xe9;
pub const SBC_ZER : u8 = 0xe5;
pub const SBC_ZRX : u8 = 0xf5;
pub const SBC_ABS : u8 = 0xed;
pub const SBC_ABX : u8 = 0xfd;
pub const SBC_ABY : u8 = 0xf9;
pub const SBC_PTX : u8 = 0xe1;
pub const SBC_PTY : u8 = 0xf1;

pub const CMP_IMM : u8 = 0xc9;
pub const CMP_ZER : u8 = 0xc5;
pub const CMP_ZRX : u8 = 0xd5;
pub const CMP_ABS : u8 = 0xcd;
pub const CMP_ABX : u8 = 0xdd;
pub const CMP_ABY : u8 = 0xd9;
pub const CMP_PTX : u8 = 0xc1;
pub const CMP_PTY : u8 = 0xd1;

pub const CPX_IMM : u8 = 0xe0;
pub const CPX_ZER : u8 = 0xe4;
pub const CPX_ABS : u8 = 0xec;

pub const CPY_IMM : u8 = 0xc0;
pub const CPY_ZER : u8 = 0xc4;
pub const CPY_ABS : u8 = 0xcc;

pub const INX     : u8 = 0xe8;
pub const DEX     : u8 = 0xca;
pub const INY     : u8 = 0xc8;
pub const DEY     : u8 = 0x88;

pub const INC_ZER : u8 = 0xe6;
pub const INC_ZRX : u8 = 0xf6;
pub const INC_ABS : u8 = 0xee;
pub const INC_ABX : u8 = 0xfe;

pub const DEC_ZER : u8 = 0xc6;
pub const DEC_ZRX : u8 = 0xd6;
pub const DEC_ABS : u8 = 0xce;
pub const DEC_ABX : u8 = 0xde;

pub const ASL_IMP : u8 = 0x0a;
pub const ASL_ZER : u8 = 0x06;
pub const ASL_ZRX : u8 = 0x16;
pub const ASL_ABS : u8 = 0x0e;
pub const ASL_ABX : u8 = 0x1e;

pub const LSR_IMP : u8 = 0x4a;
pub const LSR_ZER : u8 = 0x46;
pub const LSR_ZRX : u8 = 0x56;
pub const LSR_ABS : u8 = 0x4e;
pub const LSR_ABX : u8 = 0x5e;

pub const ROL_IMP : u8 = 0x2a;
pub const ROL_ZER : u8 = 0x26;
pub const ROL_ZRX : u8 = 0x36;
pub const ROL_ABS : u8 = 0x2e;
pub const ROL_ABX : u8 = 0x3e;

pub const ROR_IMP : u8 = 0x6a;
pub const ROR_ZER : u8 = 0x66;
pub const ROR_ZRX : u8 = 0x76;
pub const ROR_ABS : u8 = 0x6e;
pub const ROR_ABX : u8 = 0x7e;

pub const BIT_ZER : u8 = 0x24;
pub const BIT_ABS : u8 = 0x2c;

pub const TXA     : u8 = 0x8a;
pub const TAX     : u8 = 0xaa;
pub const TYA     : u8 = 0x98;
pub const TAY     : u8 = 0xa8;
pub const TXS     : u8 = 0x9a;
pub const TSX     : u8 = 0xba;

pub const CLC     : u8 = 0x18;
pub const SEC     : u8 = 0x38;
pub const CLI     : u8 = 0x58;
pub const SEI     : u8 = 0x78;
pub const CLD     : u8 = 0xd8;
pub const SED     : u8 = 0xf8;
pub const CLV     : u8 = 0xb8;

pub const NOP     : u8 = 0xea;
pub const NOP_2   : u8 = 0x1a;
pub const NOP_3   : u8 = 0x3a;
pub const NOP_4   : u8 = 0x5a;
pub const NOP_5   : u8 = 0x7a;
pub const NOP_6   : u8 = 0xda;
pub const NOP_7   : u8 = 0xfa;
pub const NOP_8   : u8 = 0x80;